### ISSUE TITLE
Forgetting learnt uncertainty reduction of liquidity in case of contradictions

### ISSUE THREAD

**Issue by renepickhardt** [lightningnetwork/lnd#7553]
While doing the information theoretic payment delivery you keep track of two values for each channel in each direction 

1. `failAmount`  is the amount that in your last observation the channel was not able to forward
2. `successAmount` is the amount that in your last observation the channel was able to forward

In the follwoing line you make a sanity check:

https://github.com/lightningnetwork/lnd/blob/51df3e5b1ea2f1f749652cecf26e7471d35aa629/routing/probability_bimodal.go#L511

if the check triggers you return a `0` probability. However I think it might make sense to just reset the learnt knowledge as it is clearly outdated to `successAmount = 0` and `failAmount = capacity` and compute the probability from there and relearn knowledge by potentially attempting a payment (depending on the result of the probability result). Of course instead of resetting both variables one could only reset the one variable for which the information is older.

btw I didn't say that before but I like the naming `failAmount` and `successAmount` better than eclairs naming `high` and `low` or my own naming `maxLiquidity` and `minLiquidity`.

**Comment by bitromortac** [comment]
I agree that this may be to exclusionary (thinking about submitting a fix), also see this comment:
https://github.com/lightningnetwork/lnd/blob/51df3e5b1ea2f1f749652cecf26e7471d35aa629/routing/probability_bimodal.go#L487

It won't fail forever however, after a certain time, the `failAmount` and `successAmount` should recover back to their default values of `successAmount=0` and `failAmount=capacity` via the exponential decays.

**Comment by renepickhardt** [comment]
oh yeah I know that the exponential decay fixes this (: It is also only a minor observation. I just thought in that case one could directly forget the outdated information. 

The reason I wanted to emphasize this is in the [wider context of weather an exponential decay is necessary at all](https://github.com/lightningdevkit/rust-lightning/pull/1227#pullrequestreview-857050321). (we don't know how strong the decay should be) thus my initial (but not necessarily better) approach would be to continue learning and once we get a contradiction forget everything for that channel and start over with no knowledge. Following that philosophy I thought I pointed out that one could directly forget the knowledge while maintaining the decay that you have.
